lib/traject/annotation_config.rb

```diff
@@ -0,0 +1,76 @@
+require 'traject_plus/macros'
+
+# rubocop:disable Style/MixinUsage
+extend TrajectPlus::Macros
+# rubocop:enable Style/MixinUsage
+
+settings do
+  provide 'processing_thread_pool', 1
+end
+
+each_record do |record, context|
+  context.clipboard[:annotation] = ::JSON.parse(record.data)
+  context.clipboard[:canvas] = ::JSON.parse(Faraday.get(record.canvas).body)
+  context.clipboard[:manifest] = ::JSON.parse(Faraday.get(context.clipboard[:annotation]['on'].first['within']['@id']).body)
+end
+
+to_field 'id', (accumulate { |resource, *_| resource.uuid })
```

Comment:
I don't think this id won't work for annotations that appear in multiple exhibits.